lib/watir-webdriver/elements/checkbox.rb

```diff
@@ -17,14 +17,7 @@ class CheckBox < Input
     #
 
     def set(bool = true)
-      assert_exists
-      assert_enabled
-
-      if @element.selected?
-        @element.click unless bool
-      else
-        @element.click if bool
-      end
+      set? == bool ? assert_enabled : click
```

Comment:
So normally I'd follow the same approach as in Radio:
click unless set?
This spec fails, because it wants to ensure enabled, even if we don't click anything:
https://github.com/watir/watirspec/blob/master/radio_spec.rb#L234